pulp_ansible/app/serializers.py

```diff
@@ -477,6 +479,54 @@ class Meta:
         model = CollectionVersion
 
 
+class CollectionVersionSignatureSerializer(SingleArtifactContentSerializer):
+    """
+    A serializer for signature models.
+    """
+
+    content = DetailRelatedField(
+        help_text=_("The content this signature is pointing to."),
+        view_name_pattern=r"content(-.*/.*)-detail",
+        queryset=CollectionVersion.objects.all(),
+    )
+    public_key = serializers.CharField(help_text=_("The public key used to verify the signature"))
+    pubkey_fingerprint = serializers.CharField(help_text=_("The fingerprint of the public key."))
```

Comment:
I think, these fields could benefit from a keyring table. Specifically if you sign all your stuff with the same key.
If we don't want to do that now, the fingerprint alone should be sufficient. Think about how to get this information in a sync workflow.

Reply:
Would we store the keyring table? Not really sure what you mean by how the fields would benefit.

I was thinking when I add the signature to the collection-version metadata, it would be added as a link to view all the signatures on that collection-version. That view would be a json list of `{'pubkey_fingerprint': ..., 'signature': 'signature_filename_link'}` for each signature. This would be what I think the CLI would use (it might still be too many requests for the CLI though). For syncing I would create a special endpoint that would aggregate all of the signatures for each collection-version into one giant json list, similar to how the `collections/all` endpoint works now.

Reply:
I was thinking, that you will end up with many collectionversions singed by the same key, and while the fingerprint is something small, the actual public key can be rather big. So duplicating the key data in the CollectionVersionSignature table seems resource hungry.
I would imagine, there would be a public-key-table (fields: fingerprint, public_key, uid, _whatever-else-interesting_, ...) and the collection would reference this table either via uuid/pk, or simply by the fingerprint. The CollectionSerializer can then annotate the signature lookup query with e.g. the `pulic_key__uid` field to provide a human readable answer to "Who signed this?".

Reply:
Re my comments in the model section, maybe we just pull the fingerprint only into this serializer via the through looking on `signing_service` ?